channels/tasks_test.py

```diff
@@ -56,19 +56,46 @@ def test_evict_expired_access_tokens():
     assert not RedditAccessToken.objects.filter(id=expired.id).exists()
 
 
-def test_subscribe_all_users_to_default_channel(settings, mocker):
+def test_subscribe_all_users_to_channels(settings, mocker, mocked_celery):
+    """Test that the main task batches out smaller tasks correctly"""
+    settings.OPEN_DISCUSSIONS_DEFAULT_CHANNEL_BACKPOPULATE_BATCH_SIZE = 2
+    users = sorted(UserFactory.create_batch(4), key=lambda user: user.username)
+    channel_names = ["nochannel_1", "nochannel_2"]
+
+    mock_subscribe_user_range_to_channels = mocker.patch(
+        "channels.tasks.subscribe_user_range_to_channels"
+    )
+
+    with pytest.raises(mocked_celery.replace_exception_class):
+        tasks.subscribe_all_users_to_channels.delay(channel_names=channel_names)
+
+    assert mocked_celery.group.call_count == 1
+    list(mocked_celery.group.call_args[0][0])
```

Comment:
what's the purpose of this `list` call?

Reply:
it's a quirk of how the mocked_celery stuff works, see this comment here (which I didn't want to proliferate everywhere): https://github.com/mitodl/open-discussions/blob/5b347f82123c5c4af04e1b02c841b7a9fed8a804/search/tasks_test.py#L254-L256

Reply:
ah, I figured it had to do with a generator or something along those lines